src/widgets/Survey/ui/ScreenManager.tsx

```diff
@@ -54,9 +58,26 @@ export const ScreenManager = ({ openTimesUpModal }: Props) => {
   ]);
 
   useEntityTimer({
-    onFinish: onEntityTimerFinish,
+    onFinish: onTimerFinish,
   });
 
+  useEffect(() => {
+    if (!autoCompletionState) {
+      return;
+    }
+
+    if (
+      autoCompletionState.activityIdsToSubmit.length ===
+      autoCompletionState.successfullySubmittedActivityIds.length
+    ) {
+      return;
+    }
+
+    if (openTimesUpModal) {
```

Comment:
Ain't no way `useEffect` should do something based on the component's props. `useEffect` is all about synchronization with the external system like Browser API (e.g.setTimeout):
https://react.dev/reference/react/useEffect
https://react.dev/learn/synchronizing-with-effects

Therefore, `openTimesUpModal` should be invoked in a function that is a result of a certain action

Reply:
Done